src/System.Reflection.Metadata.ILDasm/src/ILEventDefinition.cs

```diff
@@ -0,0 +1,174 @@
+            }
+        }
+
+        public bool HasAdder
+        {
+            get
+            {
+                return !_accessors.Adder.IsNil;
+            }
+        }
+
+        public bool HasRemover
+        {
+            get
+            {
+                return !_accessors.Remover.IsNil;
+            }
+        }
+
+        public bool HasRaiser
+        {
+            get
+            {
+                return !_accessors.Raiser.IsNil;
+            }
+        }
+
+        public ILEntity Type
+        {
+            get
+            {
+                if (!_isEntityInitialized)
+                {
+                    _isEntityInitialized = true;
+                    _type = GetEntity();
+                }
+                return _type;
+            }
+        }
+
+        public ILMethodDefinition Adder
+        {
+            get
+            {
+                if (!_isAdderInitialized)
+                {
+                    _isAdderInitialized = true;
+                    if (HasAdder)
+                    {
+                        _adder = ILMethodDefinition.Create(_accessors.Adder, ref _readers, _typeDefinition);
+                    }
+                }
+                return _adder;
+            }
+        }
+
+        public ILMethodDefinition Remover
+        {
+            get
+            {
+                if (!_isRemoverInitialized)
+                {
+                    _isRemoverInitialized = true;
+                    if (HasRemover)
+                    {
+                        _remover = ILMethodDefinition.Create(_accessors.Remover, ref _readers, _typeDefinition);
+                    }
+                }
+                return _remover;
+            }
+        }
+
+        public ILMethodDefinition Raiser
+        {
+            get
+            {
+                if (!_isRaiserInitialized)
+                {
+                    _isRaiserInitialized = true;
+                    if (HasRaiser)
+                    {
+                        _raiser = ILMethodDefinition.Create(_accessors.Raiser, ref _readers, _typeDefinition);
+                    }
+                }
+                return _raiser;
+            }
+        }
+
+        public IEnumerable<ILCustomAttribute> CustomAttributes
+        {
+            get
+            {
+                if(_customAttributes == null)
+                {
+                    _customAttributes = GetCustomAttributes();
+                }
+                return _customAttributes;
+            }
+        }
+
+        public void Accept(IVisitor visitor)
+        {
+            visitor.Visit(this);
+        }
+
+        private IEnumerable<ILCustomAttribute> GetCustomAttributes()
+        {
+            foreach(var handle in _eventDefinition.GetCustomAttributes())
+            {
+                var attribute = _readers.MdReader.GetCustomAttribute(handle);
+                yield return new ILCustomAttribute(attribute, ref _readers);
```

Comment:
We don't use yield in framework code. It's inefficient and sometimes unpredictable.